rswag-specs/spec/rswag/specs/openapi_formatter_spec.rb

```diff
@@ -0,0 +1,627 @@
+            expect(doc_2[:paths]['/path/'][:get][:parameters]).to eql([{ in: :headers }])
+            expect(doc_2[:paths]['/path/'][:get][:requestBody]).to eql(content: {
+              'application/xml' => { schema: { foo: :bar } },
+              'application/json' => { schema: { foo: :bar } }
+            })
+          end
+
+          it 'adds security to operation' do
+            expect(doc_2[:paths]['/path/'][:get][:security]).to eql([
+              {
+                my_auth: [],
+                oauth2_with_scopes: [:scope1, :scope2]
+              },
+              {
+                auth_with_this: []
+              }
+            ])
+          end
+        end
+
+        context 'with OpenAPI 3 formData' do
+          let(:doc_2) do
+            {
+              paths: {
+                '/path/' => {
+                  post: {
+                    summary: 'Retrieve Nested Paths',
+                    tags: ['nested Paths'],
+                    produces: ['application/json'],
+                    consumes: ['multipart/form-data'],
+                    parameters: [{
+                      in: :formData,
+                      schema: { type: :string }
+                    },{
+                      in: :headers
+                    }]
+                  }
+                }
+              }
+            }
+          end
+
+          it 'removes remaining consumes/produces' do
+            expect(doc_2[:paths]['/path/'][:post].keys).to eql([:summary, :tags, :parameters, :requestBody])
+          end
+
+          it 'duplicates params in: :formData to requestBody from consumes list' do
+            expect(doc_2[:paths]['/path/'][:post][:parameters]).to eql([{ in: :headers }])
+            expect(doc_2[:paths]['/path/'][:post][:requestBody]).to eql(content: {
+              'multipart/form-data' => { schema: { type: :string } }
+            })
+          end
+        end
+
+        context 'with OpenAPI 3 formData file upload' do
+          let(:doc_2) do
+            {
+              paths: {
+                '/path/' => {
+                  post: {
+                    summary: 'Upload file',
+                    produces: ['application/json'],
+                    consumes: ['image/png', 'application/octet-stream'],
+                    parameters: [
+                      {
+                        in: :formData,
+                        schema: { type: :file }
+                      }
+                    ]
+                  }
+                }
+              }
+            }
+          end
+
+          it 'generates schema in requestBody for content type' do
+            expect(doc_2[:paths]['/path/'][:post][:requestBody]).to eql(content: {
+              'image/png' => {schema: {type: :string, format: :binary}},
+              'application/octet-stream' => {schema: {type: :string, format: :binary}}
+            })
+          end
+        end
+
+        context 'with OpenAPI 3 formData file upload as part of multipart' do
```

Comment:
Since only OpenAPI v3 is now supported, I think version in `context` descriptions can be omitted

```suggestion
        context 'formData file upload as part of multipart' do
```